index.js

```diff
@@ -42,15 +51,27 @@ class TemplateImportProcessor extends BroccoliFilter {
     let imports = [];
     let rewrittenContents = contents.replace(IMPORT_PATTERN, (_, localName, importPath) => {
       if (importPath.startsWith('.')) {
-        importPath = path.resolve(relativePath, '..', importPath);
-        importPath = path.relative(this.options.root, importPath);
+        importPath = path.resolve(relativePath, '..', importPath).split(path.sep).join('/');
+        importPath = path.relative(this.options.root, importPath).split(path.sep).join('/');
       }
-      imports.push({ localName, importPath });
+      imports.push({ localName, importPath, isLocalNameValid: isValidVariableName(localName) });
       return '';
     });
 
-    let header = imports.map(({ importPath, localName }) => {
-      return `{{#let (component '${ importPath }') as |${ localName }|}}`;
+    let header = imports.map(({ importPath, localName, isLocalNameValid }) => {
+      const warnPrefix = 'ember-template-component-import: ';
+      const abstractWarn = `${warnPrefix} Allowed import variable names - CamelCased strings, like: FooBar, TomDale`;
+      const componentWarn =  `${warnPrefix}Warning! "${localName}" is not allowed as Variable name for Template import`;
+      const warn = isLocalNameValid ? '' : `
+        {{log '${componentWarn}'}}
+        {{log '${abstractWarn}'}}
+        <pre data-test-name="${localName}">${componentWarn}</pre>
+        <pre data-test-global-warn="${localName}">${abstractWarn}</pre>
+      `;
+      if (!isLocalNameValid) {
+        this._console.log(componentWarn);
```

Comment:
If we throw an error here, does it kill the ember server? If not, I'd say lean towards throwing here. If it's not a valid identifier, it _cannot_ work, so I don't see a strong use case for letting it off as a warning.

Reply:
Ah - but I imagine that is probably harder to test for?

Reply:
right, can't figure out how to test it

Reply:
added throwing for all cases, instead of testing.
if we found incorrect name in path !== 'dummy/pods/application/template.hbs' we throw.
else - we log error (and able to test throwed message)